llvm/CMakeLists.txt

```diff
@@ -624,7 +624,7 @@ option(LLVM_ENABLE_DUMP "Enable dump functions even when assertions are disabled
 option(LLVM_UNREACHABLE_OPTIMIZE "Optimize llvm_unreachable() as undefined behavior (default), guaranteed trap when OFF" ON)
 
 if( NOT uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG" )
-  option(LLVM_ENABLE_ASSERTIONS "Enable assertions" OFF)
+  option(LLVM_ENABLE_ASSERTIONS "Enable assertions" ON)
```

Comment:
Depending upon what @fhahn thinks this patch looks reasonable to me - reducing compile time even for an assertion build should help our buildbots. The results @nikic posted for the compile-time tracker looked good. I guess you'd need to revert this change first though.